buildtest/modules/util.py

```diff
@@ -59,24 +70,26 @@ def get_command(self):
 class Module:
     """Class declaration for Module class"""
 
-    def __init__(self, modules, purge=True, force=False):
+    def __init__(self, modules, purge=True, force=False, debug=False):
         """Initialize method for Module class.
 
         :param modules: list of modules
         :param purge: boolean to control whether to purge modules before loading
         :param force: boolean to control whether to force purge modules before loading
+        :param debug: debug mode for troubleshooting
 
         :type modules: list
         :type purge: bool
         :type force: bool
+        :type debug: bool
         """
+        self.debug = debug
+        self.modules = modules
 
-        # raise TypeError exception if collection is not a string type since that is required
-        # when working with module collection
+        # convert input to list if not specified already.
         if not isinstance(modules, list):
-            raise TypeError(f"Type Error: {collection} is not of type list")
+            self.modules = [modules]
 
```

Comment:
I think you would still need to do:

```
if not isinstance(modules, list):
      modules = [modules]
self.modules = modules
```
No matter what is input (list or string) you want self.modules set.

Reply:
i tried what you did, and it turns out this breaks badly. When running ``test_modules()`` you get the following output which is not right. For now it needs to be before, assumption was if its a list it is fine, otherwise it converts to a list. This logic can be revisited. I personally like throwing the ``TypeError()`` to be very explicit to on what type of argument type can be passed into `Modules` class.

```
[DEBUG] Executing module command: module purge && module load B &&  module load i &&  module load s &&  module load o &&  module load n &&  module load / &&  module load 3 &&  module load . &&  module load 0 &&  module load . &&  module load 4
```

Reply:
This suggests that you are still looping through a string (the module name) and not the list - are you sure about your implementation?